test/e2e_node/e2e_node_suite_test.go

```diff
@@ -78,15 +81,39 @@ var _ = AfterSuite(func() {
 	if e2es != nil && *startServices && *stopServices {
 		glog.Infof("Stopping node services...")
 		e2es.stop()
-		b := &bytes.Buffer{}
+	}
+})
+
+var _ Reporter = &LogReporter{}
+
+type LogReporter struct{}
+
+func (lr *LogReporter) SpecSuiteWillBegin(config config.GinkgoConfigType, summary *types.SuiteSummary) {
+	b := &bytes.Buffer{}
+	b.WriteString("******************************************************\n")
+	glog.V(2).Infof(b.String())
+}
+
+func (lr *LogReporter) BeforeSuiteDidRun(setupSummary *types.SetupSummary) {}
+
+func (lr *LogReporter) SpecWillRun(specSummary *types.SpecSummary) {}
+
+func (lr *LogReporter) SpecDidComplete(specSummary *types.SpecSummary) {}
+
+func (lr *LogReporter) AfterSuiteDidRun(setupSummary *types.SetupSummary) {}
+
+func (lr *LogReporter) SpecSuiteDidEnd(summary *types.SuiteSummary) {
+	// Only log the binary output if the suite failed.
+	b := &bytes.Buffer{}
+	if e2es != nil && !summary.SuiteSucceeded {
+		b.WriteString(fmt.Sprintf("Process Log For Failed Suite On %s\n", *nodeName))
 		b.WriteString("-------------------------------------------------------------\n")
 		b.WriteString(fmt.Sprintf("kubelet output:\n%s\n", e2es.kubeletCombinedOut.String()))
 		b.WriteString("-------------------------------------------------------------\n")
-		b.WriteString(fmt.Sprintf("apiserver output:\n%s", e2es.apiServerCombinedOut.String()))
-		b.WriteString("-------------------------------------------------------------\n")
-		b.WriteString(fmt.Sprintf("etcd output:\n%s", e2es.etcdCombinedOut.String()))
+		b.WriteString(fmt.Sprintf("apiserver output:\n%s\n", e2es.apiServerCombinedOut.String()))
 		b.WriteString("-------------------------------------------------------------\n")
-		glog.V(2).Infof(b.String())
-
+		b.WriteString(fmt.Sprintf("etcd output:\n%s\n", e2es.etcdCombinedOut.String()))
 	}
-})
+	b.WriteString("******************************************************\n")
+	glog.V(2).Infof(b.String())
```

Comment:
Sorry if I was not clear. The rationale to print logs only on failure is very sound. But why `V(2)` instead of `V(0)`?